src/rust/lib/graph-editor/src/lib.rs

```diff
@@ -1738,17 +1824,59 @@ fn new_graph_editor(world:&World) -> GraphEditor {
 
 
     // === Move Edges ===
-
-    cursor_pos_on_detach    <- cursor.frp.position.sample(&inputs.some_edge_targets_detached);
+    detached_edge           <- any(&inputs.some_edge_targets_detached,&inputs.some_edge_sources_detached);
+    cursor_pos_on_detach    <- cursor.frp.position.sample(&detached_edge);
     edge_refresh_cursor_pos <- any (cursor_pos_on_detach,cursor.frp.position);
-    eval edge_refresh_cursor_pos ([edges](position) {
+
+    is_hovering_output <- inputs.hover_node_output.map(|target| target.is_some());
+    hover_node         <- inputs.hover_node_output.unwrap();
+
+    edge_refresh_on_node_hover       <- all(edge_refresh_cursor_pos,hover_node);
+    edge_refresh_cursor_pos_no_hover <- edge_refresh_cursor_pos.gate_not(&is_hovering_output);
+
+    eval edge_refresh_cursor_pos ((position) {
         edges.detached_target.for_each(|id| {
             if let Some(edge) = edges.get_cloned_ref(id) {
                 edge.view.frp.target_position.emit(position.xy())
             }
-        })
+        });
+    });
+
+    eval edge_refresh_cursor_pos_no_hover ([edges,model](position) {
+        edges.detached_source.for_each(|edge_id| {
+            if let Some(edge) = edges.get_cloned_ref(edge_id) {
+                edge.view.frp.source_width.emit(cursor::DEFAULT_RADIUS);
+                edge.view.frp.source_height.emit(cursor::DEFAULT_RADIUS);
+                edge.view.frp.target_position.emit(-position.xy());
```

Comment:
how many times it is being evaluated? On every cursor movement, right? If so, then on every cursor movement you are emitting 3 events inside, and each event triggers full re-draw of the edge, which is pricy. There should be only 1 redraw here. Im not sure how to do it in the best way, but maybe by creating `frp.source_width_no_redraw` would be the way to go for now. If you've got better idea, id love to hear it.

Oh, below you are calling `refresh_edge_position` which does redraw again. Maybe no frp events should do redraw there and we should have redraw frp event instead on edges?

Reply:
I went with the explicit `redraw` event instead. Makes for a nicer api.